### PR TITLE
[APIC-309] [APIC-317] DOC clarify `buf` and `expires_at` for Civis file functions

### PR DESCRIPTION
**Pull request by jacksonlee-civis**
This PR updates the docstrings for the following Civis file functions:

* `file_to_civis` (for `buf` and `expires_at`)
* `dataframe_to_file` (for `expires_at`)
* `json_to_file` (for `expires_at`)

### CODE REVIEW

File: civis/io/_files.py [hunk 1]
```diff
@@ -181,8 +181,13 @@ def file_to_civis(buf, name=None, api_key=None, client=None, **kwargs):
...
     buf : file-like object or str
-        The file or other buffer that you wish to upload. Strings will be
-        treated as paths to local files to open.
+        Either a file-like object for a buffer or a string for a local file
+        path.
+        Note that if a file-like object is provided and it's not
+        an :class:`io.BufferedReader <io.BufferedReader>`
+        or :class:`io.TextIoWrapper <io.TextIoWrapper>` object,
+        the current implementation requires extra disk space
+        (which could be an issue if your file is large).
```
Comment (jacksonlee-civis) [inline civis/io/_files.py:190]: I compiled the Sphinx docs locally and verified that the formatting and links would all be good:

Screen Shot 2021-11-01 at 2 54 01 PM

File: civis/io/_files.py [hunk 2]
```diff
@@ -196,6 +201,14 @@ def file_to_civis(buf, name=None, api_key=None, client=None, **kwargs):
...
         Extra keyword arguments will be passed to the file creation
         endpoint. See :func:`~civis.resources._resources.Files.post`.
+        In particular, ``expires_at`` can be specified for
+        the date and time the file will expire. If not specified, the file
+        will expire in 30 days. To keep a file indefinitely, specify ``None``.
+        To specify a date and time, format it by the ISO 8601 standard,
+        e.g., ``"2020-12-31"``, ``"2020-12-31T23:03:40Z"``,
+        and what the ``isoformat()`` method returns from
+        a :class:`datetime.date <datetime.date>`
+        or :class:`datetime.datetime <datetime.datetime>` object.
```
Comment (jacksonlee-civis) [inline civis/io/_files.py:211]: I compiled the Sphinx docs locally and verified that the formatting and links would all be good:

Screen Shot 2021-11-01 at 2 54 08 PM

File: civis/io/_files.py [hunk 3]
```diff
@@ -495,10 +508,13 @@ def dataframe_to_file(df, name='data.csv', expires_at='DEFAULT',
...
-        expire in 30 days. To keep a file indefinitely, specify null.
-        If provided, this must be either `None` or
-        a valid RFC3339 date/Time string.
+        The date and time the file will expire. If not specified, the file
+        will expire in 30 days. To keep a file indefinitely, specify ``None``.
+        To specify a date and time, format it by the ISO 8601 standard,
+        e.g., ``"2020-12-31"``, ``"2020-12-31T23:03:40Z"``,
+        and what the ``isoformat()`` method returns from
+        a :class:`datetime.date <datetime.date>`
+        or :class:`datetime.datetime <datetime.datetime>` object.
```
Comment (jacksonlee-civis) [inline civis/io/_files.py:517]: I compiled the Sphinx docs locally and verified that the formatting and links would all be good:

Screen Shot 2021-11-01 at 2 54 20 PM

**Review by Drew-Pappas — APPROVED** [review]
LGTM

**Pull request merged**